### ISSUE TITLE
Invalid authentication - Tested on multiple accounts

### ISSUE THREAD

**Issue by MastaSlyce** [tuya/tuya-home-assistant#457]
I enter Access ID, Access Secret Mobile App and have tried Country Code both with the + and without.

Account and Password are correct, I even changed password and am able to log into the Tuya app without any issues.

I setup a new account and add that to my Home. That too gave me invalid authentication

![error](https://user-images.githubusercontent.com/55357379/132931276-d69a79d8-8740-495b-bb1b-1c20aa292573.png)

**Comment by voed** [comment]
Same problem.

**Comment by czmatt** [comment]
check you country setting in https://iot.tuya.com/user/account/info

**Comment by realjax** [comment]
Yep, same for me. Can't login anymore.  Had it working for a while but no it says 'Authentication deny' and I can no longer get it to work.

**Comment by FranMaverix** [comment]
Same here.
I have double-checked my credenteials and are all ok.
I have tried too my country (Spain) code with 'es' and '34'

Note: I have Local Tuya working in my Home Asistant.

**Comment by patrickdundas** [comment]
Also having this issue. Tried changing password, using new account, etc. I can log into the Tuya app and website but not through home assistant

**Comment by umcusumc** [comment]
In my case I already use Tuya IoT Platform some time ago and it was problem with outdated trial. Go to https://iot.tuya.com/cloud/ and next to "My Cloud Projects" you can point to "Trial edition". Actual date should be between "From" and "Expiration Time". If it's not, than you should choose "Update" button and extend trial edition for free. After this probably you should also create new cloud project according to the instalation manual.

**Comment by pluskal** [comment]
My account shows a valid trial edition subscription, but authentication issues affect me as well.

```
TRIAL EDITION
From : 2021-02-25 20:38
Expiration Time : 2022-02-26 17:59
```

**Comment by Nozza-VS** [comment]
Finally decided to set up this integration today but am also encountering authentication errors (with a valid subscription). Only this integration is affected, all other Tuya logins (Phone app/IoT Platform/LocalTuya home assistant addon/custom tuya api app) are working as expected

**Comment by marine1988** [comment]
Yes I tried everything can´t login either !

**Comment by VNRARA** [comment]
Same...

**Comment by Gessink** [comment]
same problem here. smart life app is linked, trial is valid, everything, but still authitencation error

**Comment by Aldaran** [comment]
Same... Smart Life App...

**Comment by fubar2** [comment]
Good to discover it's not just me.This is annoying. I have a bunch of tp-link switches and they are trivial with HA - this one tuya based product has been a complete time waster so far as I can't control the darn thing.

**Comment by VNRARA** [comment]
Go to Mark Watt Techs Facebook page for the solution with the beta update.

**Comment by fubar2** [comment]
> Go to Mark Watt Techs Facebook page for the solution with the beta update.

Adding a plus in front of my country code didn't help me if that's what you meant?

**Comment by VNRARA** [comment]
This comment: https://www.facebook.com/groups/markwatttech/posts/3098414410390941/?comment_id=3100223783543337&reply_comment_id=3100297030202679&__cft__[0]=AZXygc6uM5yD204XzdCvhuTWWz6RNnwsjK7P7N8w535B7HYPnaEjfz0xOZd1YnKsLdbXXC61xNPTnELy0eQLmGlrzEWc_PUM7KksYf13EwyrUwtQ-nARj9f6kBMtT1IH9NU&__tn__=R]-R

**Comment by fubar2** [comment]
PSA: Country had to be the full name - for me Australia instead of the country code 61 worked with the very latest github code..
